### DISCUSSION TITLE
PCB HAT for phoniebox

### DISCUSSION THREAD

**Discussion by Surfer2010** [MiczFlor/RPi-Jukebox-RFID#1805] [Ideas]
since a while ago i wanted to create pcb hat design for the raspberry gpio pins which connects all together and is the only thing you need.

goal is to have a pcb hat which does the following:
1. On/Off Shim attachable by special pin headers
2. HifiBerry same as on/off shim ... headers to attach it
3. number of buttons with/without leds in it as screw terminals to attach the cables of the buttons
4. Optional LEDs connectors to have custom leds usable

i know (1) and (2) can be done by extension boards for the raspberry but adding buttons it is a lot of manual work to do instead of having screw terminals for all the rest

what do you think, is there anyone willing to help knowing about pcb design?

is there by any chance a pcb design available like the gpio extension boards to edit and don't need to start from scratch?

**Comment by s-martin** [comment]
Übung hab ich leider keine, aber ich fänds super!

**Comment by seisfeld** [comment]
Wir (wir sind die Jungs vom TonUINO Projekt, lieben aber auch andere Musikboxprojekte) haben uns mit ein paar Leuten zusammen gesetzt und vor einiger Zeit so etwas in der Richtung entwickelt. Es ist ein HAT der eigentlich alles vereint. Shim, Codec, Verstärker, diverse Anschlüsse. Im Prinzip braucht man nur noch den Raspi, MFRC522, Buttons und Lautsprecher. Die erste Inkarnation ist in kleiner Stückzahl [hier](https://www.leiterkartenpiraten.de/produkt/lkp-audiohatcl/) zu bekommen, eine kleine Variante (im Zero Formfaktor) ist bereits in Arbeit. Wir sind auch bereit Samples an Phoniebox Entwickler zu schicken falls Interesse besteht. Einen [Thread dazu gibts in unserem Forum](https://discourse.voss.earth/t/audio-hat-cl/11239) auch. Wir würden uns freuen wenn wir die Community bereichern könnten. :)

**Reply by s-martin** [reply]
Very cool, I would be interested ;)

Nevertheless I need to check, which Raspi I currently use in my Phoniebox, not sure if it's a 3A 🤔

**Reply by seisfeld** [reply]
Der HAT funktioniert im Prinzip mit jedem neueren Raspi. Beim 4er sind halt zwei Anschlüsse (Mic und Line In) an der Seite nicht nutzbar, dies ist aber bei einer Phoniebox zu vernachlässigen. Melde dich am besten im Forum an und schreib mir dort ne PN (an stephan), dann können wir alles weitere bzgl. Adresse pi pa po besprechen.

**Reply by s-martin** [reply]
Hi, Anmeldung hat funktioniert, PN senden leider nicht...

**Reply by seisfeld** [reply]
Ich hab dein Userlevel erhöht, sollte jetzt gehen.

**Comment by obehr** [comment]
Hello @Surfer2010 and others ,

by coincidence I had the same idea :-) Today I finished a first draft of my PCB and then noticed this thread. I think I got most of your requirements. I did not want any extra cards so I included all circuits on the PCB. The audio part is not really HiFi but said to be ok.

- DAC (PCM5102a)
- Amplifier (PAM8406)
- LED open drain buffers
- Button inputs with hardware debounce
- Screw clamps (2,54 mm grid) for connecting button and LEDs
- Pin header for RFID-RC522
- Reverse engineered and slightly modified OnOffSHIM circuit

Have a look at the Github page: https://github.com/obehr/phoniebox_hat

I simulated the OnOffSHIM circuit in LTSpice and it seems to work. Yet everything is completely untested. So there might be bugs in the schematic. I'm also not completely shure, how to connect the DAC to the amplifier. I used the schematics from application notes.

Any comments are appreciated. Also some sharp eyes, that could have a look at the schematic. We could make this an equivalent to the Tonuino HAT but I do not make it more complicated as it needs to be ;-)

Kind regards

Oliver

![Screenshot of PCB](https://github.com/obehr/phoniebox_hat/blob/main/doc/PCB.png?raw=true)

**Reply by obehr** [reply]
Update: Added I2C header for displays. Reworked debounce circuits.

**Reply by Surfer2010** [reply]
Here is such a board aswell 
https://www.leiterkartenpiraten.de/produkt/lkp-audiohatcl/

**Reply by obehr** [reply]
Oh, thanks for the info! Its a nice board with all essential features and ready to order.

Pros:
- Mic and Line in
- Dedicated rotary encoder connector (you could use the button inputs with my board instead)

Cons:
- The output drivers have only 1 W per speaker. I wanted to upgrade from 3 W to 5 W because the volume was a bit low.
- WM8960 seems to have lower SNR than the PCM5102a (would not overrate this)
- I prefer screw clamps to connect the arcade buttons in my box.
- No I2C header for display
- The positioning of the line inputs is not according to the official HAT spec. You might get some collision when not using RPi 3B+.

So for me it makes sense to continue with my board (I would've liked to build it anyway). I might order assembled PCBs at JLCPCB within the next four weeks.
